### PR TITLE
Add moduledoc option to defmock

### LINKED ISSUE #73: Feature request: option to add @moduledoc false
The mock module may be defined in the environment where `mix docs` runs for some reasons. In that case modules generated by `defmock` will be included in the final doc.

Can we add an option to skip moduledoc? something like this:

```elixir
Mox.defmock(MyApp.CalcMock, for: MyApp.Calculator, moduledoc: false)
```

### PR DESCRIPTION
**Pull request by chulkilee**
Closes #73

### CODE REVIEW

File: lib/mox.ex [hunk 1]
```diff
@@ -293,11 +301,15 @@ defmodule Mox do
       end
 
     skip_optional_callbacks = Keyword.get(options, :skip_optional_callbacks, [])
+    moduledoc = Keyword.get(options, :moduledoc, nil)
```
[the remark this answers is not in the document]
Comment (chulkilee) [reply lib/mox.ex]: Changed to `false`.

**Comment by josevalim** [comment]
It looks good but CI is failing for earlier Elixir versions. We need to run
that test only on 1.7 (iirc) and later.
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by josevalim** [comment]
Ping! :)

**Comment by chulkilee** [comment]
> It looks good but CI is failing for earlier Elixir versions. We need to run that test only on 1.7 (iirc) and later.

Could you give me example or hint how I can do that? (I asked that question on existing thread, but probably it was hidden because that was on "outdated" change)

**Comment by josevalim** [comment]
@chulkilee oops, sorry! I think wrapping the test in a `function_exported?` check would be enough:

    if function_exported?(Code, :fetch_docs, 1) do

**Comment by chulkilee** [comment]
Done - I found existing code to exclude tests based on `System.version/0` and `Version.match?/2` - so used it.

File: test/test_helper.exs [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
-  if Version.match?(System.version(), ">= 1.8.0") do
-    []
-  else
-    [:requires_caller_tracking]
-  end
+  [
+    {"< 1.7.0", [:requires_code_fetch_docs]},
+    {"< 1.8.0", [:requires_caller_tracking]}
+  ]
+  |> Enum.reduce([], fn {version, tags}, acc ->
```
Comment (josevalim) [inline test/test_helper.exs]: You can use flat_map:

```elixir
Enum.flat_map(fn {version, tags} ->
      if Version.match?(System.version(), version), do: tags, else: []
end)
```

Can you please amend accordingly? Thank you!

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**